### ISSUE TITLE
Custom leap with s/S will get remapped after the first jump

### ISSUE THREAD

**Issue by ShaolunWang** [ggandor/leap.nvim#191]
As title suggested, if I map any key that is one of the default keys, e.g. `s` to bidirectional leap it only work once:
```lua
vim.keymap.set({ 'n', 'x', 'o' }, 's', function()
	local current_window = vim.fn.win_getid()
	require('leap').leap({ target_windows = { current_window } })
end, { noremap = true, desc = 'leap' })
```

And after leaping for the first time `s` would get remapped back to the default vim one. This might be caused by executing unbinding during every leap

**Comment by ggandor** [comment]
I cannot reproduce, the problem is not with this mapping itself, it should work.

**Comment by ShaolunWang** [comment]
Right, thanks for confirming that!

**Comment by ShaolunWang** [comment]
I just narrowed down the issue I had,fyi this is triggered by setting it like this:
```lua
require('lazy').setup({
  {
    'ggandor/leap.nvim',
    keys = {'s'} 
    -- with this keys lazy loading option,
    -- at each leap it calls lua/leap/main.lua: line 1202
    -- it also depends on the 
    -- and it will get reset to default vim s key
  }
})

vim.keymap.set({ 'n', 'x', 'o' }, 's', function()
  local current_window = vim.fn.win_getid()
  require('leap').leap({ target_windows = { current_window } })
end, { noremap = true})
```
Despite said [here](https://github.com/ggandor/leap.nvim#lazy-loading) in the readme that lazy loading isn't necessary, there would potentially still be people like me who interpret the wording as lazy loading leap wouldn't result in unexpected behavior(s). Would it be possible to state this somewhere?

**Comment by ggandor** [comment]
I don't fully understand the problem (what Lazy is doing in the background), and cannot really follow the comment in your code snippet, so please walk me through this then.

**Comment by ShaolunWang** [comment]
So the problem I had was lazy would reset the keybinds when I use the lazy load on keypress option, if I decided to disable the default keymaps and only map `s/S` to leap actions. Those keybinds get reset to the vim s/S after the first leap action

**Comment by Sleepful** [comment]
did you try this?
```lua
require('lazy').setup({
  {
    'ggandor/leap.nvim',
    keys = {
        "s",
        function()
          require('leap').leap()
        end,
        mode = { "x", "n" },
        desc = "Leap",
      }
  }
})
```

**Comment by ShaolunWang** [comment]
@Sleepful This would only work with the first leap. On the second leap it shows:
```
E5108: Error executing lua: ~/.local/share/nvim/lazy/leap.nvim/lua/leap/main.lua:405: attempt to index local '_local_67_' (a nil value)
stack traceback:
        ~/.local/share/nvim/lazy/leap.nvim/lua/leap/main.lua:405: in function 'leap'
        ~/.config/nvim/lua/plugins/leap.lua:11: in function <~/.config/nvim/lua/plugins/leap.lua:10>
        ```

**Comment by cd-a** [comment]
Did anyone find a solution to this?

**Comment by ggandor** [comment]
TL;DR: don't use `keys`, it's unnecessary.
